src/QtLocationPlugin/QGCTileCacheWorker.cpp

```diff
@@ -116,61 +116,19 @@ QGCCacheWorker::run()
         _init();
     }
     if(_valid) {
-        _db = new QSqlDatabase(QSqlDatabase::addDatabase("QSQLITE", kSession));
-        _db->setDatabaseName(_databasePath);
-        _db->setConnectOptions("QSQLITE_ENABLE_SHARED_CACHE");
-        _valid = _db->open();
+        _connectDB();
     }
     _deleteBingNoTileTiles();
+    QMutexLocker lock(&_taskQueueMutex);
     while(true) {
         QGCMapTask* task;
         if(_taskQueue.count()) {
```

Comment:
This is the call to `_taskQueue.count()` which was racey prior to this PR.

![QGCCacheWorker _taskQueue is racey](https://user-images.githubusercontent.com/72767521/114422467-eb8b7500-9b7b-11eb-9cf2-663e77fec3f1.png)